### DISCUSSION TITLE
Upstream source of the (MS?) Office stencil set?

### DISCUSSION THREAD

**Discussion by davebarkerxyz** [jgraph/drawio#3379] [Q&A]
Does anyone know the origin of the Office stencil set (found at https://github.com/jgraph/drawio/tree/dev/src/main/webapp/stencils/office)? I've looked through the commit history for this project and for mxgraph but I'm having trouble finding out where those icons originated. I expect they're from Microsoft as one featured the MS Office logo.

I understand that the usage terms at https://www.diagrams.net/doc/faq/usage-terms state ...

> The diagrams you create belong to you and we grant you a license to use any of our copyrighted icons in your diagrams and for making raster and vector versions of your diagrams and parts, using diagrams.net, for any purpose.

... but I'd like to ascertain that those stencils into our internal (and potentially external) documentation, and particularly in case we need to extract them to use them in other contexts (e.g. on our documentation site, but not as part of a diagram created with Diagrams.net/Draw.io).

Thanks,
Dave

**Comment by davebarkerxyz** [comment]
After a little more digging, it looks like they came from https://www.microsoft.com/en-us/download/details.aspx?id=35772

I found a post at https://modery.net/updated-free-visio-stencils-for-office-365-exchange-lync-and-sharepoint-2015-edition/ with some more details. However, I couldn't find any license information which could be problematic if our information governance unit queries this. I'm not sure they're compatible with the Apache 2.0 license as they seem to lack clear licensing terms upstream (though I'd be happy to be wrong here).

**Comment by davidjgraph** [comment]
The bulk comes from https://learn.microsoft.com/en-us/azure/architecture/icons/. They document as:

```
Do’s
• Use the icons to illustrate how products can work together
• In diagrams, we recommend to include the product name somewhere close to the icon
• Use the icons as they would appear within Azure
Don’ts
• Don’t crop, flip or rotate icons
• Don’t distort or change icon shape in any way
• Don’t use Microsoft product icons to represent your product or service
Terms
Microsoft permits the use of these icons in architectural diagrams, training materials, or documentation. You may copy, distribute, and display the icons only for the permitted use unless granted explicit permission by Microsoft. Microsoft reserves all other rights.
```

We do perform as much due diligence as we reasonably can on all third-party stencils where we are not entirely the authors. The original license on these was pretty awful, so we asked MS to remove parts, https://github.com/MicrosoftDocs/architecture-center/issues/2251.

**Comment by davebarkerxyz** [comment]
Hi David,

That's really helpful - thanks! It does seem that the stencil license is in conflict with the application's Apache 2.0 license (i.e. if I clone from GitHub and look at the LICENSE file, it would be forgivable to assume that this covered all of the application content). However, I do absolutely understandable the reasoning behind the license discrepancy, and I appreciate that you reached out to Microsoft regarding the previous license wording.

I wonder if it might be possible to surface stencil licensing terms somewhere? It seems like if I were to use the upper right icon in the [example linked](https://learn.microsoft.com/en-us/azure/architecture/solution-ideas/media/advanced-analytics-on-big-data.png), horizontally flip it and refer to it as "Network processor" (instead of Azure Analysis Services) then I would be using it otherwise in accordance with the linked General Guidelines and consequently *may* be breaking the stencil license terms. I do note that the README.md has a section *Logo and trademark usage* which specifically covers the draw.io trademark and logos. Might a subsequent section (*Third party stencil license / usage*) be a suitable place to mention terms of the stencil licenses? The "Office" stencil category doesn't actually mention Microsoft explicitly - I wasn't sure if they were produced by Microsoft or by a third party.

Cheers,
Dave

**Reply by davidjgraph** [reply]
There is an additional license section in the main README https://github.com/jgraph/drawio. IANAL, but the do's and don'ts don't appear under the terms section. The text is not remotely legalese, even MS would have a hard time enforcing if someone flipped an icon.

Could you maybe help me understand the background for the request, i.e. how are you using the project and why would you need such a high level of due diligence?

**Reply by davebarkerxyz** [reply]
I agree that the Microsoft guidelines aren’t explicitly worded and may not even be enforceable as license terms (if that’s how they are intended to be interpreted), so in a practical sense there’s probably little risk here.

I work at a large organisation (a university) where we are expected to be able to demonstrate the origin and license of any IP that we use (both software and other IP like design elements that might appear in our docs and comms) should we be internally or externally audited. Until this thread, if I used the “Office” stencils, all I could have said was that they came from draw.io, who’s maintainers say that, regarding third party icons and stencils, “the original license allows use in this project”.

Personally, I think it’s valuable to be able to discover the licensing terms for all the elements of an open source project.

That said, I don’t think it’s a big issue (which is why I raised it as a discussion thread) - initially I just wanted to know “if I use these icons, that I believe to be third party ones, on our docs site, internal comms or transformed in some way on a poster at an outreach event (i.e. use of the elements outside draw.io by transforming the outputted diagram), would I be breaching any license terms I might not be able to easily discover”.

I do appreciate that you’ve taken the time to share the source of these icons and chat over some of the licensing points.
